Generic_Environment.php

```diff
@@ -207,4 +209,87 @@ public function is_advanced_cache_add_in() {
 		return ( ( $script_data = @file_get_contents( W3TC_ADDIN_FILE_ADVANCED_CACHE ) )
 			&& strstr( $script_data, 'PgCache_ContentGrabber' ) !== false );
 	}
+
+	/**
+	 * Enables crawling blocker for cache directory
+	 *
+	 * @throws Util_WpFile_FilesystemOperationException with S/FTP form if it can't get the required filesystem credentials
+	 */
+	private function robots_add_directive() {
+		$robots_path = Util_Environment::site_path() . 'robots.txt';
+
+		$robots_content = @file_get_contents( $robots_path );
+		if ( $robots_content === false )
+			return;
+
+		$new_robots_content = $this->robots_remove_from_content( $robots_content );
+		$new_robots_content = $new_robots_content . $this->robots_addon();
+
+		if ( $new_robots_content != $robots_content ) {
+			try {
+				Util_WpFile::write_to_file( $robots_path, $new_robots_content );
+			} catch ( Util_WpFile_FilesystemOperationException $ex ) {
+				throw new Util_WpFile_FilesystemModifyException(
+					$ex->getMessage(), $ex->credentials_form(),
+					'Edit file <strong>' . $robots_path .
+					'</strong> and add next lines:', $robots_path,
+					$this->robots_addon() );
```

Comment:
WPCS: One argument per line for multi-line function/method calls.  Closing parenthesis is to be on a new line `);`.
The message to be displayed should be translatable.